clippy_lints/src/misc_early/mod.rs

```diff
@@ -318,6 +319,34 @@ declare_clippy_lint! {
     "tuple patterns with a wildcard pattern (`_`) is next to a rest pattern (`..`)"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for `[all @ ..]` patterns.
+    ///
+    /// ### Why is this bad?
+    /// In all cases, `all` works fine and can often make code simpler, as you possibly won't need
+    /// to convert from say a `Vec` to a slice by dereferencing.
+    ///
+    /// ### Example
+    /// ```rust,ignore
+    /// if let [all @ ..] = &*v {
+    ///     // NOTE: Type is a slice here
+    ///     println!("all elements: {all:#?}");
+    /// }
+    /// ```
+    /// Use instead:
+    /// ```rust,ignore
+    /// if let all = v {
+    ///     // NOTE: Type is a `Vec` here
+    ///     println!("all elements: {all:#?}");
+    /// }
```

Comment:
It warns `irrefutable_let_patterns` in this case, so can we add the case that simply print?

It would be better to detect `irrefutable_let_patterns`, but IMO this lint is still worth it as is.
```suggestion
    /// if let all = v {
    ///     // NOTE: Type is a `Vec` here
    ///     println!("all elements: {all:#?}");
    /// }
    /// // or
    /// println!("all elements: {v:#?}");
```

Reply:
I think this is fine. Many lints clippy or not will still lint even if it'll give a warning, and then that lint will hopefully be applied as well by the programmer. Though yeah I'm fine with adding just the `println!`